### PR TITLE
Added new package langium-sprotty

### PR DESCRIPTION
**Pull request by spoenemann**
This is work in progress for integrating Langium with Sprotty. I will test it with the "states" example in [sprotty-vscode](https://github.com/eclipse/sprotty-vscode).

### CODE REVIEW

**Comment by spoenemann** [comment]
I will add an example to [sprotty-vscode](https://github.com/eclipse/sprotty-vscode) that demonstrates how this works. But I cannot add that example before this is merged and a snapshot version is published.

File: packages/langium/src/dependency-injection.ts
Comment (rainerschuster) [inline packages/langium/src/dependency-injection.ts:36]: Add JavaDoc
```suggestion
 * @param module4 (optional) fourth Module
 * @returns a new object of type I
```

Comment (rainerschuster) [inline packages/langium/src/dependency-injection.ts:36]:
```suggestion
 * @param module4 (optional) fourth Module
 * @returns a new object of type I
```

**Review by rainerschuster — CHANGES_REQUESTED** [review]
Fix JavaDoc for inject

File: packages/langium-sprotty/src/diagram-server-manager.ts
Comment (danieldietrich) [inline packages/langium-sprotty/src/diagram-server-manager.ts:32]: (not only for documentation purposes) it would be better, if `DefaultDiagramServerManager` would implement `DiagramServerManager`

File: packages/langium-sprotty/src/lsp.ts
Comment (danieldietrich) [inline packages/langium-sprotty/src/lsp.ts:15]: Side-note: I ask myself why we do lint namespaces away, they seem to be useful for us. Or is there a better alternative? Maybe we should just allow namespaces (part of another PR).

File: packages/langium/src/dependency-injection.ts
Comment (danieldietrich) [inline packages/langium/src/dependency-injection.ts:38]: _(ah - I spotted the fourth module 🙂, maybe I will be able to massage the types as part of my DI efforts, the way that we allow N modules)_

**Review by danieldietrich — CHANGES_REQUESTED** [review]
Well done, Sir! Please find my review comments. Thx!

File: packages/langium/src/documents/document-builder.ts
Comment (msujew) [inline packages/langium/src/documents/document-builder.ts:172]: We do this quite often throughout our codebase. Would it make sense to add a `MultiMap` to our utilities? Doesn't have to happen in this PR, but we should create an issue for this.

**Review by msujew — COMMENTED** [review]
Aside from the issues the others have already mentioned, this looks good to me. I have a general question, but otherwise still looks good to go after addressing the other issues.

File: packages/langium/src/documents/document-builder.ts
Comment (danieldietrich) [reply packages/langium/src/documents/document-builder.ts:172]: Maybe one 'utility' function would be sufficient?

```ts
// collections.ts
export function insert<K, V>(map: Map<K, V[]>, key: K, value: V): Map<K, V[]> {
    map.has(key) ? map.get(key).push(value) : map.set(key, [value]);
    return map;
}
```

Comment (msujew) [reply packages/langium/src/documents/document-builder.ts:172]: Well, that leaves the issue of the `get` function to return `undefined` instead of an empty array, which would be preferable.

Comment (danieldietrich) [reply packages/langium/src/documents/document-builder.ts:172]: Makes sense - a `MultiMap` type would be nice.

File: packages/langium/src/documents/document-builder.ts
Comment (msujew) [inline packages/langium/src/documents/document-builder.ts:154]: Should we pass the filtered documents here, instead of all documents? Some of the documents have been indexed previously, but will still be passed to the callback. The same applies to the call of `notifyBuildPhase` of `runCancelable`.

**Review by msujew — APPROVED** [review]
The API looks good to me! I have a question about the semantics of `onBuildPhase` and which documents we pass in there.

File: packages/langium/src/utils/collections.ts
Comment (danieldietrich) [inline packages/langium/src/utils/collections.ts:66]: I had to look twice until I spotted the readonly :)

**Review by danieldietrich — APPROVED** [review]
This looks great, Miro!

File: packages/langium/src/utils/collections.ts
Comment (spoenemann) [reply packages/langium/src/utils/collections.ts:66]: I used `ReadonlyArray<V>` first, but eslint complained about that :-)

**Comment by spoenemann** [comment]
I adapted the code to the new multi-language support (#311). This required considerable changes. I will test it next week.

**Pull request merged**
